components/nimbus/src/cirrus.udl

```diff
@@ -0,0 +1,183 @@
+namespace cirrus {};
+
+dictionary AppContext {
+    string app_name;
+    string app_id;
+    string channel;
+    string? app_version = null;
+    string? app_build = null;
+    string? architecture = null;
+    string? device_manufacturer = null;
+    string? device_model = null;
+    string? locale = null;
+    string? os = null;
+    string? os_version = null;
+    string? android_sdk_version = null;
+    string? debug_tag = null;
+    i64? installation_date = null;
+    string? home_directory = null;
+    JsonObject? custom_targeting_attributes = null;
+};
+
+dictionary TargetingAttributes {
+    AppContext app_context;
+    boolean is_already_enrolled;
+    StringHashSet active_experiments;
+    string? language = null;
+    string? region = null;
+    i32? days_since_install = null;
+    i32? days_since_update = null;
+};
```

Comment:
Something I noticed with setting defaults for struct members — it doesn't seem to matter in which order you include the member variables. Regardless of the order in the UDL UniFFI seems to handle the translations appropriately, which is important since right now the UniFFI Python bindings need all members/params with defaults to be after all non-default ones (Python's positional vs named parameters ordering rules).